.github/run_erblint.sh

```diff
@@ -2,7 +2,11 @@
 
 shopt -s globstar
 
-bundle exec erblint decidim**/app/{cells,views}/**/*.erb
+DEFAULT_BRANCH=$(git symbolic-ref refs/remotes/origin/HEAD | sed 's@^refs/remotes/origin/@@')
```

Comment:
I believe we can get the base ref through the `$GITHUB_BASE_REF` environment variable here?
https://docs.github.com/en/actions/learn-github-actions/environment-variables

Note that env variable is only available for pull requests. As a fallback, you could use `$GITHUB_REF` which would also be available for the pushes and tags.

Particularly if this script is supposed to be used outside of the GitHub actions, the `HEAD` may not always exist, see:
https://stackoverflow.com/questions/17639383/how-to-add-missing-origin-head-in-git-repo/17639471#17639471

Another question about this is that when we merge this, I believe this would compare the `develop` branch against `develop` which would cause the ERBLint to be skipped always for the current build. Is this intentional?

If you need to use `git symbolic-ref`, I believe you don't need the `sed` command if you use the `--short` option:
https://git-scm.com/docs/git-symbolic-ref

Reply:
> $GITHUB_BASE_REF, $GITHUB_REF

If we move the script out of .github folder, I presume all these variables won't be available, will they?

Reply:
> If you need to use git symbolic-ref, I believe you don't need the sed command if you use the --short option:
https://git-scm.com/docs/git-symbolic-ref

I just did a copy&paste to fetch that value. But I see that the `--short` option returns `origin/develop`, unlike `sed` option which returns only `develop`. With such value, _git diff_ returns different amount of files cc/ @entantoencuanto

Reply:
> would cause the ERBLint to be skipped always for the current build

I don't get that. The script does not compare develop against develop, but the current branch against develop. Only it's skipped when no ERB changes

Reply:
> If we move the script out of .github folder, I presume all these variables won't be available, will they?

Yep, you are correct - if you run that locally, these variables won't exist.

I think you could configure the script with an ENV variable e.g. `ERBLINT_DIFF_BRANCH`.

And then from in the GitHub action, you could pass the correct value to that variable as in `${GITHUB_BASE_REF:-$GITHUB_REF}`

> I just did a copy&paste to fetch that value. But I see that the --short option returns origin/develop, unlike sed option which returns only develop.

If you compare against `develop`, it is comparing against the `develop` branch on your local machine. But if you compare against `origin/develop`, it compares it against that branch at the remote (most often this repo at GitHub but not always).

But like I said, this may be problematic in some occasions, e.g. on my setup where I'm not pointing origin to `decidim/decidim`.

> > would cause the ERBLint to be skipped always for the current build
> 
> I don't get that. The script does not compare develop against develop, but the current branch against develop. Only it's skipped when no ERB changes

I just meant the following situation:
1. We merge this at `develop`.
2. There is another PR that changes e.g. `decidim-core/app/views/layouts/decidim/_logo.html.erb`
3. In that PR everything runs fine
4. We merge that PR to `develop`
5. Now what happens?

My current understanding is that in that situation this script at `develop` would state that there are no changes, so it doesn't need to run ERBLint against any files.

So do we want the so called "current build" to always skip ERBLint totally or do we want to run it against all files when PRs are merged?

Please correct me if I am wrong with this. I may understand this part incorrectly too.

Reply:
> My current understanding is that in that situation this script at develop would state that there are no changes

:exploding_head: Mmmm, no, once merged, in the other PR the test would state: there's an erb file changed! therefore the test will run: `bundle exec erblint decidim-core/app/views/layouts/decidim/_logo.html.erb`. It will run only over that file, no against all views (modified and not modified), as currently does.

Reply:
Yes, I understand that.

But what happens after that **other PR** is merged?

I understand that ERBLint would be skipped in the merge commit.

Reply:
Still I don't get it. What happens where? In the moment this PR will be merged, the following PRs (currently open or the future ones), will be run the erblint scoped, that is, only the affected files.

I don't see which is the importance here of the other PRs, sorry.

Reply:
> the following PRs (currently open or the future ones), will be run the erblint scoped, that is, only the affected files.

Yes, this is essentially just a question how we want to do that.

> I don't see which is the importance here of the other PRs, sorry.

All I am trying to say that this approach works just fine for all the PRs. I'm not talking about PRs.

It is just a question whether we want to skip ERBLint for the "current build", i.e. not any PRs. Direct commits e.g. to `develop` (which is when we merge a PR to develop).

That's the only potential "problem" that I would see with this approach. It's just a question how we want to do that.

Reply:
Well, it's supposed to all ERB views in develop fulfill the tests. And the PR you're trying to merge has fulfilled the tests as well. Check them out again is a double check.

As alternative, the script could ask for the current_branch, and run the defaults if current_branch === default_branch. Defaults will be the previous glob pattern. It could do a `git diff current_branch..HEAD~1` also, to test against the previous commit, but this will be the same as the PR verifications you're trying to merge did before.

Reply:
> But like I said, this may be problematic in some occasions, e.g. on my setup where I'm not pointing origin to decidim/decidim.

What's this output `git symbolic-ref refs/remotes/origin/HEAD | sed 's@^refs/remotes/origin/@@'` for your mainio fork?

Reply:
> What's this output `git symbolic-ref refs/remotes/origin/HEAD | sed 's@^refs/remotes/origin/@@'` for your mainio fork?

It's "master".

Reply:
> As alternative, the script could ask for the current_branch, and run the defaults if current_branch === default_branch. Defaults will be the previous glob pattern. It could do a `git diff current_branch..HEAD~1` also, to test against the previous commit, but this will be the same as the PR verifications you're trying to merge did before.

I think this would solve the problem I am referring to.

In GitHub, I think this situation is equivalent to the `$GITHUB_BASE_REF` env variable being empty.

Reply:
For the time being, merges against develop will run through all the files